tasks/sealights/sealights-nodejs/0.1/sealights-nodejs.yaml

```diff
@@ -0,0 +1,152 @@
+        - use
+        - $(params.source-artifact)=/opt/app-root/src
+    - name: sealights-nodejs-instrumentation
+      image: registry.access.redhat.com/ubi8/nodejs-$(params.nodejs-version):latest
+      workingDir: /opt/app-root/src
+      securityContext:
+        runAsUser: 0
+      env:
+        - name: COMPONENT
+          value: $(params.component)
+        - name: SCM
+          value: $(params.scm)
+        - name: SCM_PROVIDER
+          value: $(params.scm-provider)
+        - name: REPOSITORY_URL
+          value: $(params.repository-url)
+        - name: BRANCH
+          value: $(params.branch)
+        - name: REVISION
+          value: $(params.revision)
+        - name: PULL_REQUEST_NUMBER
+          value: $(params.pull-request-number)
+        - name: TARGET_BRANCH
+          value: $(params.target-branch)
+        - name: TEST_EVENT
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['pipelinesascode.tekton.dev/event-type']
+        - name: WORKSPACE_PATH
+          value: $(params.workspace-path)
+      args: ["$(params.exclude[*])"] 
+      script: |
+        #!/bin/sh
+        # TODO: REMOVE
+        set -x
+        set -euo pipefail
+        export SEALIGHTS_TOKEN BUILD_NAME PATHS_EXCLUDED_ENUM
+        SEALIGHTS_TOKEN="$(cat /usr/local/sealights-credentials/token)"
+        BUILD_NAME="${REVISION}_$(date +'%y%m%d.%H%M')"
+        PATHS_EXCLUDED_ENUM="$(IFS=,; echo "$(printf "%s," $@ | sed 's/,$//')")"
+        # Download jq in order to help determine the Agent version below
+        curl -Lso jq https://github.com/stedolan/jq/releases/download/jq-1.7.1/jq-linux64 && chmod +x jq && mkdir -p /opt/app-root/bin && mv jq /opt/app-root/bin/jq
+        # Installs Sealights Node.js agent.
+        # Fetch Agent version defined in Sealights dashboard settings 
+        SL_VERSION=$(curl -X GET "https://redhat.sealights.co/api/v2/agents/slnodejs/recommended" -H "accept: application/json" -H "Authorization: Bearer $SEALIGHTS_TOKEN" -L | jq -r .agent.version)
+        # Install Sealights Agent (Recommended version)
+        npm i -g slnodejs@$SL_VERSION
+        # In case build is coming from a /retest event then it's a pull_request
+        # if event type is not push and pull_request_number is not empty then is a pull_request
+        if [ "$TEST_EVENT" != "push" ] && [ -n "$PULL_REQUEST_NUMBER" ]; then
+          TEST_EVENT="pull_request"
+        fi
+        # Configures the app for sealights using vars from your pipeline run.
+        # TODO: REMOVE
+        export NODE_DEBUG=sl
+        if [[ -n "${TEST_EVENT}" && "${TEST_EVENT}" == "pull_request" ]]; then
+          echo "[INFO] Generate build session id from a Pull Request source code"
+          npx slnodejs prConfig --appname "${COMPONENT}" --targetBranch "${TARGET_BRANCH}" \
+            --pullRequestNumber "${PULL_REQUEST_NUMBER}" --latestCommit "${REVISION}" --repositoryUrl "${REPOSITORY_URL}" --token "${SEALIGHTS_TOKEN}"
```

Comment:
i'm seeing my PULL_REQUEST_NUMBER and REPOSITORY_URL variables empty in my testing and I'm using the same approach you are here.

Are you seeing these vars appear on your end?

Reply:
solved in a slack thread (solution is to use PaC variables):
```
        - name: repository-url
          value: '{{ repo_url }}'
        - name: pull-request-number
          value: '{{ pull_request_number }}'
```